include/qvs.hpp

```diff
@@ -0,0 +1,174 @@
+#ifndef QVS_HPP
+#define QVS_HPP
+
+#include <limits>
```

Comment:
Make sure that all these include statements are necessary. If something is not just remove it